android/app/src/main/java/org/uwaterloo/subletr/pages/individualchat/IndividualChatPageView.kt

```diff
@@ -0,0 +1,177 @@
+				navHostController = viewModel.navHostController,
+			)
+			LazyColumn(
+				modifier = Modifier
+					.weight(weight = 1.0f),
+				verticalArrangement = Arrangement.Bottom,
+			) {
+				items(items = uiState.chatItems) { chatItem: ChatItemModel ->
+					if (chatItem is ChatItemModel.MyChatItem) {
+						MyChatBubble(myChatItem = chatItem)
+					}
+					else if (chatItem is ChatItemModel.OtherChatItem) {
+						OtherChatBubble(otherChatItem = chatItem)
+					}
+				}
+			}
+			Box(
+				modifier = Modifier
+					.wrapContentSize()
+					.clip(
+						RoundedCornerShape(
+							topStart = dimensionResource(id = R.dimen.s),
+							topEnd = dimensionResource(id = R.dimen.s),
+							bottomEnd = dimensionResource(id = R.dimen.zero),
+							bottomStart = dimensionResource(id = R.dimen.zero),
+						)
+					),
+			) {
+				Row(
+					modifier = Modifier
+						.background(secondaryBackgroundColor)
+						.fillMaxWidth(fraction = 1.0f),
+					verticalAlignment = Alignment.CenterVertically,
+					horizontalArrangement = Arrangement.Center,
+				) {
+					RoundedTextField(
+						modifier = Modifier.padding(dimensionResource(id = R.dimen.m)),
+						value = uiState.message,
+						onValueChange = {
+							viewModel.messageStream.onNext(it)
+						},
+						placeholder = {
+							Text(text = stringResource(id = R.string.message_ellipses))
+						},
+						colors = TextFieldDefaults.colors(
+							unfocusedContainerColor = primaryBackgroundColor,
+							focusedContainerColor = primaryBackgroundColor,
+							unfocusedIndicatorColor = Color.Transparent,
+							focusedIndicatorColor = Color.Transparent,
+						),
+					)
+
+					IconButton(
+						modifier = Modifier,
+						onClick = {},
+					) {
+						Icon(
+							painter = painterResource(
+								id = R.drawable.send_solid_black_24,
+							),
+							contentDescription = stringResource(id = R.string.back_arrow),
+							tint = sendButtonColor,
+						)
+					}
+				}
+			}
+		}
+	}
+}
+
+@Preview(showBackground = true)
+@Composable
+fun IndividualChatPageViewLoadingPreview() {
+	SubletrTheme {
+		IndividualChatPageView(
+			modifier = Modifier,
+			viewModel = IndividualChatPageViewModel(
+				navigationService = NavigationService(
+					context = LocalContext.current,
+				),
+			),
+			uiState = IndividualChatPageUiState.Loading,
+		)
+	}
+}
+
+@Preview(showBackground = true)
+@Composable
+fun IndividualChatPageViewLoadedPreview() {
+	SubletrTheme {
+		IndividualChatPageView(
+			modifier = Modifier,
+			viewModel = IndividualChatPageViewModel(
+				navigationService = NavigationService(
+					context = LocalContext.current,
+				),
+			),
+			uiState = IndividualChatPageUiState.Loaded(
+				basicInfo = IndividualChatPageUiState.BasicInfo(
+					contactName = "Abhed Shwarma",
+					address = "123 University Ave.",
+				),
+				chatItems = listOf(
+					ChatItemModel.MyChatItem("my test message"),
+					ChatItemModel.OtherChatItem("other test message")
+				),
```

Comment:
Let me see how quickly I can get chat working so that we can merge without hard coding strings.

Reply:
With respect to the hardcoded strings that currently exist, I think it's fine since you can't navigate to this page anyways.
Like, this page isn't visible so there isn't any way that a potential user could see this.